pymc3/distributions/distribution.py

```diff
@@ -141,26 +159,102 @@ def __new__(cls, name, *args, **kwargs):
                 "for a standalone distribution."
             )
 
-        rng = kwargs.pop("rng", None)
+        if not isinstance(name, string_types):
+            raise TypeError(f"Name needs to be a string but got: {name}")
 
+        # Pop out PyMC3-related kwargs so only the disttribution kwargs remain
+        rng = kwargs.pop("rng", None)
         if rng is None:
             rng = model.default_rng
 
-        if not isinstance(name, string_types):
-            raise TypeError(f"Name needs to be a string but got: {name}")
-
         data = kwargs.pop("observed", None)
-
         total_size = kwargs.pop("total_size", None)
+        testval = kwargs.pop("testval", None)
+        transform = kwargs.pop("transform", UNSET)
 
+        shape = kwargs.pop("shape", None)
         dims = kwargs.pop("dims", None)
+        size = kwargs.pop("size", None)
+
+        # Raise on unsupported parametrization
+        if shape is not None and dims is not None:
+            raise ValueError("Passing both `shape` ({shape}) and `dims` ({dims}) is not supported!")
+        if dims is not None and size is not None:
+            raise ValueError("Passing both `dims` ({dims}) and `size` ({size}) is not supported!")
+        if shape is not None and size is not None:
+            raise ValueError("Passing both `shape` ({shape}) and `size` ({size}) is not supported!")
+
+        # Warn about discouraged parametrization
+        if shape is not None and not isinstance(shape, (list, tuple)):
+            warnings.warn("The `shape` parameter should be a list or tuple.", UserWarning)
+            shape = (shape,)
+        if dims is not None and not isinstance(dims, (list, tuple)):
+            warnings.warn("The `dims` parameter should be a list or tuple.", UserWarning)
+            dims = (dims,)
+        if size is not None and not isinstance(size, (list, tuple)):
+            warnings.warn("The `size` parameter should be a list or tuple.", UserWarning)
+            size = (size,)
+
+        if size is not None and Ellipsis in size:
+            raise ValueError("The `size` parameter cannot contain an Ellipsis. Actual: {size}")
+        if not _valid_ellipsis_position(shape):
+            raise ValueError(
```

Comment:
This is an excessive amount of parameter validation.  If it's absolutely necessary, it should be delegated to a validation function/method.